### PR TITLE
Feature/poptarts

### PR DESCRIPTION
**Pull request by dudash**
adding popup for use location sharing

### CODE REVIEW

**Comment by dudash** [comment]
BTW - looks like I didn't break it. Mapbox might have an issue with Chrome.
Even their example doesn't work for me:
https://docs.mapbox.com/mapbox-gl-js/example/locate-user/

File: lifeline/src/components/HomePage.js [hunk 1]
```diff
@@ -59,20 +60,37 @@ function HomePage({ stateStore, coords }) {
...
-            console.log("no incident")
-            // TODO: error handling, alert user to an issue - not that they can do anything about it
-        } else {
-            try {
-                const { status } = await postMyLocation(stateStore.incidentId, latitude, longitude);
-                console.log("postMyLocation get status = " + JSON.stringify(status));
-            } catch (e) {
-                console.error('postMyLocation error!');
-                // TODO: error handling, Indicate error condition to UI
+    const popupSnackbar = async () => {
```
Comment (jkeam) [inline lifeline/src/components/HomePage.js:63]: This function doesn't need to be `async`.

File: lifeline/src/components/HomePage.js [hunk 2]
```diff
@@ -40,6 +40,7 @@ function HomePage({ stateStore, coords }) {
         // TODO: share
     };
 
+    // Not currently used
```
Comment (jkeam) [inline lifeline/src/components/HomePage.js:43]: Delete dead code over commenting out.

File: lifeline/src/components/HomePage.js (hunk 1 above)
Comment (jkeam) [inline lifeline/src/components/HomePage.js]: This function doesn't need to be `async`.

File: lifeline/src/components/HomePage.js [hunk 3]
```diff
@@ -59,20 +60,37 @@ function HomePage({ stateStore, coords }) {
...
+        }
+
+        setOpenSnackbar(false);
+    };
+
+    const shareMyLocation = async () => {        
+        if (coords) {
+            const { latitude, longitude } = coords;
+            if (!stateStore.incidentId) {
+                console.log("no incident")
```
Comment (jkeam) [inline lifeline/src/components/HomePage.js]: Nit missing semicolon.  Consider removing unneeded log statements.

File: lifeline/src/components/HomePage.js [hunk 4]
```diff
@@ -59,20 +60,37 @@ function HomePage({ stateStore, coords }) {
...
+                try {
+                    const { status } = await postMyLocation(stateStore.incidentId, latitude, longitude);
+                    console.log("postMyLocation get status = " + JSON.stringify(status));
+                } catch (e) {
+                    console.error('postMyLocation error!');
+                    // TODO: error handling, Indicate error condition to UI
+                }
             }
+        } else {
+            console.log("no coords available - need to alert user and request location")
```
Comment (jkeam) [inline lifeline/src/components/HomePage.js]: Nit missing semicolon.  Consider removing unneeded log statements.

File: lifeline/src/components/HomePage.js [hunk 5]
```diff
@@ -144,18 +151,60 @@ function HomePage({ stateStore, coords }) {
                             </Tab> */}
                         </Tabs>
                         <br></br>
```
Comment (jkeam) [inline lifeline/src/components/HomePage.js:153]: Nit, consider self closing `<br />`

File: lifeline/src/components/LocationDetailsView.js [hunk 6]
```diff
@@ -2,22 +2,25 @@
 // license: Apache 2.0
 import React from "react";
 import { observer } from "mobx-react";
-import { postMyLocation } from "../api/emergencyapi";
+// import { postMyLocation } from "../api/emergencyapi";
```
Comment (jkeam) [inline lifeline/src/components/LocationDetailsView.js:5]: Consider deletion over commenting out.

File: lifeline/src/components/LocationDetailsView.js [hunk 7]
```diff
@@ -2,22 +2,25 @@
...
 import Moment from 'react-moment';
 import 'moment-timezone';
 
 function LocationDetailsView({ stateStore }) {
-    const [mystatus, setMyStatus] = React.useState({});
-    const sendMyLocation = async (lat,lon) => {
-        const response = await postMyLocation(lat,lon);
-        setMyStatus(response.data);
-    };
+    // const sendMyLocation = async (lat,lon) => {
```
Comment (jkeam) [inline lifeline/src/components/LocationDetailsView.js:12]: Consider deletion over commenting out.

File: lifeline/src/components/LocationDetailsView.js [hunk 8]
```diff
@@ -2,22 +2,25 @@
...
+    //     const response = await postMyLocation(lat,lon);
+    //     setMyStatus(response.data);
+    // };
 
     React.useEffect(() => {
-        // TODO: tell the erdemo backend and get status
-        //sendMyLocation(stateStore.lat, stateStore.lon);
-    }, [stateStore.lat, stateStore.lon, stateStore.llUpdatedAt, stateStore.address]);
+        // TODO: uncomment to automatically send location on every change
```
Comment (jkeam) [inline lifeline/src/components/LocationDetailsView.js:18]: Consider deletion over commenting out.

File: lifeline/src/components/MapView.js [hunk 9]
```diff
@@ -40,11 +40,13 @@ function MapView({ stateStore }) {
             // on the map to make this obvious to the user
             // this feature will allow the user to mark a more precise location manually.
             // map.on("move", function() {
-            //     console.log("A move event occurred " + map.getCenter());
             //     var center = map.getCenter();
-            //     stateStore.setLat(center.lat);
-            //     stateStore.setLon(center.lng);
-            //  // TODO display animated center pin and turn off mapbox's geolocate if it's on
+            //     //if (geolocate is turned off) {
```
Comment (jkeam) [inline lifeline/src/components/MapView.js:44]: Consider deletion over commenting out.

File: yarn.lock [hunk 10]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (jkeam) [inline yarn.lock:1]: Don't check this file in.  There is no `package.json` at the root of this project.

**Review by jkeam — COMMENTED** [review]
Also general comment, but I noticed there were no tests for any of the components :( 

I'll generally do React snapshot testing, so at the least you'll know the components render, and at best, you can see what's changed whenever snapshots get regenerated and checked into source control.

https://jestjs.io/docs/en/tutorial-react#snapshot-testing

File: yarn.lock (hunk 10 above)
Comment (dudash) [reply yarn.lock:1]: maybe we should chat about this one

**Pull request merged**